mm2src/coins/lightning/ln_events.rs

```diff
@@ -124,46 +162,88 @@ fn sign_funding_transaction(
     Transaction::try_from(signed).map_to_mm(|e| OpenChannelError::ConvertTxErr(e.to_string()))
 }
 
+async fn save_channel_closing_details(
+    persister: Arc<LightningPersister>,
+    platform: Arc<Platform>,
+    user_channel_id: u64,
+    reason: String,
+) -> SaveChannelClosingResult<()> {
+    persister.update_channel_to_closed(user_channel_id, reason).await?;
+
+    let channel_details = persister
+        .get_channel_from_sql(user_channel_id)
+        .await?
+        .ok_or_else(|| MmError::new(SaveChannelClosingError::ChannelNotFound(user_channel_id)))?;
+
+    let closing_tx_hash = platform.get_channel_closing_tx(channel_details).await?;
+
+    persister
+        .add_closing_tx_to_sql(user_channel_id, closing_tx_hash)
```

Comment:
> The first thing that comes to mind is to use a mutex

https://github.com/KomodoPlatform/atomicDEX-API/blob/79982e259cdebd9c6eb7a7fec96522c097458be2/mm2src/coins/lightning_persister/src/lib.rs#L1038
isn't the SQLite connection's mutex enough?

Reply:
Hmm, my comment seems to be related to other places where we load something from the storage, change it and uploads the changed data to the storage. And the storage state may have changed between `get_..._from_sql` and `add_..._to_sql`. For example, at the `handle_payment_received` function.

Reply:
> Hmm, my comment seems to be related to other places where we load something from the storage, change it and upload the changed data to the storage.

For payments, this isn't a problem as payments are atomic in nature, so for inbound payments, either `PaymentReceived` or `PaymentFailed` event get handled for each payment. Same for outbound payments, either `PaymentSent` or `PaymentFailed` event gets handled for each payment.

For channels, I can now see that there is a very small chance of a problem if electrums are down, cause once they are back a `SpendableOutputs` event may get fired before this loop resolves https://github.com/KomodoPlatform/atomicDEX-API/blob/152b2afafa5d8926aa7bd561e2ec433786b02600/mm2src/coins/lightning/ln_platform.rs#L539  which will cause this to fail https://github.com/KomodoPlatform/atomicDEX-API/blob/152b2afafa5d8926aa7bd561e2ec433786b02600/mm2src/coins/lightning/ln_events.rs#L462-L468 since there is no closing transaction yet. This has a very small probability ofcourse since electrums have to be down for more than 6 blocks to get a `SpendableOutputs` event for a closed channel but maybe solving this with a mutex is a good idea.